src/odemis/driver/technolution.py

```diff
@@ -1,117 +1,185 @@
+        :return: content dictionary(getting), or entire response (raw_response=True)
         """
         logging.debug("Executing: %s" % url)
-        if method == _METHOD_GET:
-            resp = requests.get(url, json=data, timeout=timeout)
-        elif method == _METHOD_POST:
-            resp = requests.post(url, json=data, timeout=timeout)
+        resp = self._session.get(self._server_url + url, json=data, timeout=timeout)
 
         if resp.status_code != expected_status:
             raise AsmApiException(url, resp, expected_status)
 
         logging.debug("Call to %s went fine, no problems occured\n" % url)
+        if raw_response:
+            return resp
+        else:
+            return json.loads(resp.content)
+
+    def ASM_API_Post_Call(self, url, expected_status, data=None, raw_response=False, timeout=600):
+        """
+        Call to the ASM API to post data to the ASM API
 
+        :param url (str): url of the command, server part is defined in object variable self._server_url
+        :param expected_status (int): expected feedback of server for a positive call
+        :param data: data (request body) added to the call to the ASM server (mega field metadata, scan location etc.)
+        :param raw_response (bool): specified the format of the structure returned
+        :param timeout (int): [s] if within this period no bytes are received an timeout exception is raised
+        :return: status_code(int) or entire response (raw_response=True)
+        """
+        logging.debug("Executing: %s" % url)
+        resp = self._session.post(self._server_url + url, json=data, timeout=timeout)
+
+        if resp.status_code != expected_status:
+            raise AsmApiException(url, resp, expected_status)
+
+        logging.debug("Call to %s went fine, no problems occurred\n" % url)
         if raw_response:
             return resp
-        elif method == _METHOD_POST:
+        else:
             return resp.status_code
-        elif method == _METHOD_GET:
-            return json.loads(resp.content)
 
 
 class EBeamScanner(model.Emitter):
+    """
+    Represents the e-beam scanner of a single field image.
+    """
 
     def __init__(self, name, role, parent, **kwargs):
-        super(EBeamScanner, self).__init__(name, role, **kwargs)
-        self.parent = parent
+        """
+        Initialize the e-beam scanner of a single field image..
 
-        # self._metadata[model.MD_PIXEL_SIZE] =
+        :param name:
+        :param role:
+        :param parent:
+        :param kwargs:
+        """
+        super(EBeamScanner, self).__init__(name, role, parent=parent, **kwargs)
+
+        clockFrequencyData = self.parent.ASM_API_Get_Call("/scan/clock_frequency", 200)
+        # Check if clockFrequencyData holds the proper key
+        if 'frequency' not in clockFrequencyData:
+            raise IOError("Could not obtain clock frequency, received data does not hold the proper key")
```

Comment:
If this cannot be found most likely the API changed or the data was received incorrect, this is more check to prevent unclear error messages.
The second part is there to point the user in a direction of what is wrong. But if you like we could delete that part.

Reply:
I am not sure you even need this check. The call in line 167 should return already an error if it cannot retrieve the clock frequency. 

If you are actually able to execute line 167 without an error (please confirm) but then the key is not available, you should already get an error saying that the "key" is not available, when you want to access the value in line 171. So no need for this extra check.

Reply:
Agreed to remove. If the API changes we should make sure to adapt our wrapper.